src/main/java/io/nats/client/impl/PushMessageManager.java

```diff
@@ -171,7 +188,15 @@ && extractFcSubject(msg) == null)
         return msg;
     }
 
+    protected boolean subManage(Message msg) {
+        return false;
+    }
+
     boolean manage(Message msg) {
+        if (!sub.getSID().equals(msg.getSID())) {
+            return true;
+        }
```

Comment:
Combined the sid check as part of sub classing change. Not really required except for ordered, but will come in handing for future management that might replaced the subscription under the covers.